content/teknologi/altinnstudio/changelog/app-nuget/v4/whats-new.md

```diff
@@ -5,10 +5,34 @@ description: Overview of changes introduced in version 4
 toc: true
 ---
 
+### 4.7.0 (2021-06-08)
+
+Altinn Apps now authorize access for statless apps.
+
+Altinn Apps have now two new Application Events where application developers can add data processing logic. Calulation, population +++
+
+In addition, RunCalculate Application event is now OBSOLETE. It is recommended  that Apps should move from RunCalculation to RunProcessDataWrite and RunProcessDataRead.
+In the future, RunCalculate methods will not be supported.
+
+The process to update is
+
+1. Add DataProcessing folder and Handler from template [here](https://github.com/Altinn/altinn-studio/tree/master/src/Altinn.Apps/AppTemplates/AspNet/App/logic)
+2. Update App.cs. Add referane to dataprocessinghandler and copy new methods from [App.cs](https://github.com/Altinn/altinn-studio/blob/master/src/Altinn.Apps/AppTemplates/AspNet/App/logic/App.cs#L112)
```

Comment:
Will these line number references be future proof? Won't we risk that things might be added and line number don't apply?

Reply:
fixed